### PR TITLE
DEVP-3725 resume feed

### PR DESCRIPTION
**Pull request by MickStanciu**
## Summary

This PR will solve 2 items:
* make the action-timelines-items resumable
* dedupe action-timelines-items

## Checklist

- [x] Is this PR a reasonable size?

- [ ] List deployment sequence with all relevant PRs.
  - ...

- [ ] For breaking changes: I have confirmed that other repos are not affected, or I have planned a deployment sequence to address any impact.

---

**Code Review Guidelines** for Reviewers

- Try to review in a timely manner. Opinions/nitpicks should not be blockers. Pair on a call for non-trivial feedback.
- Overall design and approach should follow established patterns. Don't try to make the PR perfect.
- Try to identify edge cases, race conditions, over-engineering, lack of test coverage and complexity.
- If you don't feel qualified to review the code, pass it on to someone who is.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,11 +1,11 @@
 module github.com/SafetyCulture/safetyculture-exporter
 
-go 1.22
+go 1.23
 
-toolchain go1.22.3
+toolchain go1.24.1
```
Comment (sc-gv) [inline go.mod:5]: is it okay to use a different toolchain version?

Comment (MickStanciu) [reply go.mod:5]: that's automatically by go. I cannot touch that. Go will put it back

File: pkg/internal/feed/feed_action_timeline_item.go [hunk 2]
```diff
@@ -6,6 +6,7 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/MickStanciu/go-fn/fn"
```
Comment (sc-gv) [inline pkg/internal/feed/feed_action_timeline_item.go:9]: do we really need a dependency here? 

([A little copying is better than a little dependency.](https://go-proverbs.github.io/ ))

Comment (MickStanciu) [reply pkg/internal/feed/feed_action_timeline_item.go:9]: we are using go-fn anyway. The de-dupe fn is in that lib

**Review by sc-gv — APPROVED** [review]

File: pkg/internal/feed/feed_action_timeline_item.go [hunk 3]
```diff
@@ -95,14 +96,24 @@ func (f *ActionTimelineItemFeed) Export(ctx context.Context, apiClient *httpapi.
...
 	drainFn := func(resp *GetFeedResponse) error {
 		var rows []*ActionTimelineItem
 
 		if err := json.Unmarshal(resp.Data, &rows); err != nil {
 			return fmt.Errorf("map data: %w", err)
 		}
 
-		if len(rows) != 0 {
+		// deduplicate rows (hotfix) because the feed returns duplicates and this creates PK violations issues
+		deDupedRows := fn.DeduplicateList(rows, func(row *ActionTimelineItem) string {
```
Comment (HarryTranAU) [inline pkg/internal/feed/feed_action_timeline_item.go:112]: shouldn't we be deduping at the feed level?

File: go.mod (hunk 1 above)
Comment (MickStanciu) [reply go.mod:5]: yes
